javascript/ql/src/semmle/javascript/security/UselessUseOfCat.qll

```diff
@@ -163,10 +171,12 @@ module PrettyPrintCatCall {
         callback = createCallbackString(cat.getCallback())
         or
         callback = "" and not exists(cat.getCallback())
-      )
+      ) and
+      fileArg = createFileArgument(cat).trim() and
+      not(containsNonTrivialBashChar(fileArg.regexpReplaceAll("\\$|\\`| ", ""))) // string concat might contain " ", template strings might contain "$" or `, and that is OK. 
```

Comment:
```suggestion
     // sanity check in case of surprising `toString` results, other uses of `containsNonTrivialBashChar` should ensure that this conjunct will hold most of the time
      not(containsNonTrivialBashChar(fileArg.regexpReplaceAll("\\$|\\`| ", ""))) // string concat might contain " ", template strings might contain "$" or `, and that is OK. 
```